demo/src/lib/layout/iframe.ts

```diff
@@ -0,0 +1,96 @@
+import type {UnsubscribeFunction, UnsubscribeObject} from '@amadeus-it-group/tansu';
+import {asReadable, writable} from '@amadeus-it-group/tansu';
+
+/**
+ * Creates an iframe handler, allowing to enable dynamic resizing and detect if the parent should display a spinner.
+ * The returned spinner store includes a debounce for better user experience.
+ *
+ * In order to detect when the page is fully loaded, we listen to message events.
+ *
+ * @param defaultHeight the default height in pixels of the iframe
+ * @param resize enable dynamic resizing
+ * @param messageType the type of the message event we listen to
+ * @param spinnerDebounce the debounce in milliseconds before the spinner store is set to true
+ * @returns the handler and a spinner store
+ */
+export function createIframeHandler(defaultHeight: number, resize = true, messageType = 'sampleload', spinnerDebounce = 300) {
+	const _iframeLoaded$ = writable(true);
+	const _showSpinner$ = writable(false);
+	const _height$ = writable(0);
+
+	let spinnerTimer: any;
+	let resizeObserver: ResizeObserver | undefined;
+	let heightSubscription: (UnsubscribeFunction & UnsubscribeObject) | undefined;
+
+	const setupObserver = (iframe: HTMLIFrameElement) => {
```

Comment:
As discussed, this should be extracted in a reactive utility (similar to the intersection utility) that could also be used in the slider.

Reply:
Tracked in #291